### PR TITLE
When failing to send to new identity, save it.

### PR DESCRIPTION
**Pull request by michaelkirk**
Remove creation of "old style" blocking SN alerts

PTAL @charlesmchen

### CODE REVIEW

File: src/Messages/InvalidKeyMessages/TSInvalidIdentityKeySendingErrorMessage.m [hunk 1]
```diff
@@ -43,20 +43,12 @@ - (instancetype)initWithOutgoingMessage:(TSOutgoingMessage *)message
...
-                                                                                forRecipient:recipientId
-                                                                                preKeyBundle:preKeyBundle];
-    return message;
-}
-
 - (void)acceptNewIdentityKey
 {
+    // Shouldn't really get here, since we're no longer creating blocking SN changes.
+    // But there may still be some old unaccepted SN errors in the wild that need to be accepted.
+    OWSFail(@"accepting new identity key is deprecated.");
```
Comment (charlesmchen) [inline src/Messages/InvalidKeyMessages/TSInvalidIdentityKeySendingErrorMessage.m:50]: Can you make sure this is in the test plan?

File: src/Messages/OWSMessageSender.m [hunk 2]
```diff
@@ -885,20 +858,47 @@ - (void)sendMessage:(TSOutgoingMessage *)message
...
-            [[TSInvalidIdentityKeySendingErrorMessage
-                untrustedKeyWithOutgoingMessage:message
-                                       inThread:thread
-                                   forRecipient:exception.userInfo[TSInvalidRecipientKey]
-                                   preKeyBundle:exception.userInfo[TSInvalidPreKeyBundleKey]] save];
+            // This *can* happen under normal usage, but it should happen relatively rarely.
+            // We expect it to happen whenever Bob reinstalls, and Alice messages Bob before
+            // she can pull down his latest identity.
+            // If it's happening a lot, we should rethink our profile fetching strategy.
+            OWSAnalyticsInfo(@"Message send failed due to untrusted key.");
```
Comment (charlesmchen) [inline src/Messages/OWSMessageSender.m:865]: Nice - I got to get in the habit of using these macros.

File: src/Messages/TSMessagesManager.m [hunk 3]
```diff
@@ -1001,8 +1001,12 @@ - (void)processException:(NSException *)exception envelope:(OWSSignalServiceProt
       } else if ([exception.name isEqualToString:InvalidVersionException]) {
           errorMessage = [TSErrorMessage invalidVersionWithEnvelope:envelope withTransaction:transaction];
       } else if ([exception.name isEqualToString:UntrustedIdentityKeyException]) {
-          errorMessage =
-              [TSInvalidIdentityKeyReceivingErrorMessage untrustedKeyWithEnvelope:envelope withTransaction:transaction];
```
Comment (charlesmchen) [inline src/Messages/TSMessagesManager.m:1005]: You've deprecated `TSInvalidIdentityKeySendingErrorMessage` above by removing its initializer. 

* Can we deprecate `TSInvalidIdentityKeyReceivingErrorMessage` too?
* Can we mark the `TSErrorMessageWrongTrustedIdentityKey` enum value as "unused"?

**Review by charlesmchen — APPROVED** [review]
LGTM

File: src/Messages/TSMessagesManager.m (hunk 3 above)
Comment (michaelkirk) [reply src/Messages/TSMessagesManager.m:1005]: I'll leave a note to the effect of: "We no longer create these, but legacy ones continue to exist"

File: src/Messages/InvalidKeyMessages/TSInvalidIdentityKeySendingErrorMessage.m (hunk 1 above)
Comment (michaelkirk) [reply src/Messages/InvalidKeyMessages/TSInvalidIdentityKeySendingErrorMessage.m:50]: done!

**Pull request merged**